frontend/src/v5/ui/routes/dashboard/viewerCanvases/viewerCanvases.component.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ViewerCanvas as Canvas3D } from '@/v4/routes/viewerCanvas';
+import { DrawingViewer as Canvas2D } from '@components/viewer/drawingViewer/drawingViewer.component';
+import { useLocation } from 'react-router-dom';
+import { SplitPane } from './viewerCanvases.styles';
+import { ViewerCanvasesContext } from '../../viewer/viewerCanvases.context';
+import { useCallback, useContext, useState } from 'react';
+import { clamp } from 'lodash';
+
+export const ViewerCanvases = () => {
+	const MIN_PANEL_SIZE = 68;
+	const windowWidth = window.innerWidth;
+	const { pathname } = useLocation();
+	const { is2DOpen, leftPanelRatio, setLeftPanelRatio } = useContext(ViewerCanvasesContext);
+	const [size, setSize] = useState(windowWidth * leftPanelRatio);
+
+	const handleWindowResize = useCallback(({ currentTarget: { innerWidth } }) => {
+		const newSize = clamp(leftPanelRatio * innerWidth, MIN_PANEL_SIZE, innerWidth - MIN_PANEL_SIZE);
+		setSize(newSize);
+	}, [leftPanelRatio]);
+	window.addEventListener('resize', handleWindowResize);
+
+	const onDragResize = (s: number) => {
+		setSize(s);
+		setLeftPanelRatio(s / window.innerWidth);
+	};
+
+	return (
+		<SplitPane
+			split="vertical"
+			size={is2DOpen ? size : '100%'}
+			minSize={MIN_PANEL_SIZE}
+			maxSize={windowWidth - MIN_PANEL_SIZE}
+			onChange={onDragResize}
+		>
+			<Canvas3D location={{ pathname }} />
```

Comment:
Maybe instead of having this canvas3d is better to have all the ui inside. 
Would be more consistent with the way the 2d part works.
Also  we could change the name to 3dviewer and 2dviewer beacause is the whole component,

Reply:
I've put the toolbar inside the 3D Viewer component. I didn't put the panels in their as they shouldn't be affected by whether the 2d is open or not (i.e. right panels are still on the right side of the screen)